dbms/src/Interpreters/Context.cpp

```diff
@@ -569,6 +571,19 @@ ConfigurationPtr Context::getUsersConfig()
     return shared->users_config;
 }
 
+void Context::setSecurityConfig(Poco::Util::AbstractConfiguration & config, const LoggerPtr & log)
+{
+    LOG_INFO(log, "Setting secuirty config.");
+    auto lock = getLock();
+    shared->security_config = std::make_shared<TiFlashSecurityConfig>(config, log);
+}
+
+TiFlashSecurityConfigPtr Context::getSecurityConfig()
+{
+    auto lock = getLock();
```

Comment:
Maybe the lock is useless?

Reply:
No. You still need that lock to prevent read/write race of the `shared_ptr`.

Reply:
BTW, thread-safety of `shared_ptr` is in the fact that multiple threads can use multiple `shared_ptr`s pointing to the same object and don't have to worry about the race when multiple threads try to destruct the same object. Other than that, there is no thread-safety about `shared_ptr`, esp. for the case that one thread tries to assign a `shared_ptr` an object and the other is using that `shared_ptr`.